diff --git a/comfy/ops.py b/comfy/ops.py
index a1a1a1a1..b2b2b2b2 100644
--- a/comfy/ops.py
+++ b/comfy/ops.py
@@ -67,7 +67,11 @@ try:
         if "set_priority" in inspect.signature(sdpa_kernel).parameters:
             SDPA_BACKEND_PRIORITY = [
                 SDPBackend.FLASH_ATTENTION,
-                SDPBackend.CUDNN_ATTENTION,
+                # This leads to a crash since we are lacking some dep in nixpkgs
+                # however, according to Connor from CUDA team, this would only be useful
+                # to people with data center GPUs so we're not loosing anything by commenting
+                # it out.
+                # SDPBackend.CUDNN_ATTENTION,
                 SDPBackend.EFFICIENT_ATTENTION,
                 SDPBackend.MATH,
             ]

